services/src/main/java/org/exoplatform/news/service/impl/NewsTargetingServiceImpl.java

```diff
@@ -196,6 +202,36 @@ public Metadata updateNewsTargets(String originalTargetName, NewsTargetingEntity
     return metadataService.updateMetadata(storedMetadata, userIdentityId);
   }
 
+  @Override
+  public List<NewsTargetingEntity> getTargetsByUser(String userName) {
+    org.exoplatform.services.security.Identity identity = NewsUtils.getUserIdentity(userName);
+    List<String> publisherPermissions = new ArrayList<>();
+    List<NewsTargetingEntity> targetingEntities = getTargets();
+    List<String> publisherGroups = identity.getGroups()
+                                           .stream()
+                                           .filter(group -> identity.isMemberOf(String.valueOf(group), PUBLISHER_MEMBERSHIP_NAME))
+                                           .collect(Collectors.toList());
+    publisherPermissions.addAll(publisherGroups);
+    try {
+      List<Space> spaces = spaceService.getAccessibleSpaces(userName);
+      List<String> publisherSpaces = spaces.stream()
+                                           .filter(space -> spaceService.isPublisher(space, userName))
+                                           .map(space -> SPACE_PREFIX + space.getDisplayName())
+                                           .collect(Collectors.toList());
+      publisherPermissions.addAll(publisherSpaces);
+
+    } catch (SpaceException e) {
+      LOG.error("An error occurred when trying to retrieve the list of accessible spaces ", e);
+    }
+
+    targetingEntities = targetingEntities.stream()
```

Comment:
can we here just verify if each target permission matches with a publisher membership of the user ?